### PR TITLE
Remove p8platform

### PR DESCRIPTION
**Pull request by phunkyfish**
v6.0.2
- Remove p8-platform dependency
- Remove p8-platform OS includes
- Use chrono to get millis time
- Use std::mutex
- Use kodi StringUtils
- Remove SAFE_DELETE
- Remove p8-platform time utils
- Use chrono for timing
- Use std::thread
- Include windows.h for LARGE_INTEGER, MAX_PATH and GetTempAthA support
- add temporary workaround about attribute hidden

### CODE REVIEW

**Comment by phunkyfish** [comment]
Quite a lot of change here so will need a proper review.

Can you also runtime test @AlwinEsch ?

File: src/KeepAliveThread.cpp [hunk 1]
```diff
@@ -23,22 +22,45 @@ CKeepAliveThread::CKeepAliveThread(cPVRClientArgusTV& instance) : m_instance(ins
...
+
+void CKeepAliveThread::Process()
 {
   kodi::Log(ADDON_LOG_DEBUG, "CKeepAliveThread:: thread started");
-  while (!IsStopped())
+  while (m_running)
   {
     int retval = m_instance.GetRPC().KeepLiveStreamAlive();
     kodi::Log(ADDON_LOG_DEBUG, "CKeepAliveThread:: KeepLiveStreamAlive returned %i", (int)retval);
-    // The new P8PLATFORM:: thread library has a problem with stopping a thread that is doing a long sleep
```
Comment (AlwinEsch) [inline src/KeepAliveThread.cpp:35]: Think this also present by way new here and inside Kodi it was done by:
```cpp
  void Sleep(uint32_t milliseconds)
  {
    if (milliseconds > 10 && IsCurrentThread())
    {
      std::unique_lock<std::recursive_mutex> lock(m_threadMutex);
      m_stopEvent.wait_for(lock, std::chrono::milliseconds(milliseconds));
    }
    else
    {
      std::this_thread::sleep_for(std::chrono::milliseconds(milliseconds));
    }
  }
```
To stop direct after stopping signal is come

File: src/lib/tsreader/MultiFileReader.cpp [hunk 2]
```diff
@@ -43,8 +41,9 @@
 
 #include <kodi/Filesystem.h>
 #include <kodi/General.h>
+#include <kodi/tools/EndTime.h>
 
-using namespace P8PLATFORM;
+//using namespace P8PLATFORM;
```
Comment (AlwinEsch) [inline src/lib/tsreader/MultiFileReader.cpp]: This line can be then deleted.

File: src/lib/tsreader/MultiFileReader.cpp [hunk 3]
```diff
@@ -398,7 +397,8 @@ long MultiFileReader::RefreshTSBufferFile()
       // try to clear local / remote SMB file cache. This should happen when we close the filehandle
       m_TSBufferFile.CloseFile();
       m_TSBufferFile.OpenFile();
-      usleep(5000);
+      std::this_thread::sleep_for(std::chrono::milliseconds(5));
+
```
Comment (AlwinEsch) [inline src/lib/tsreader/MultiFileReader.cpp]: No new line needed here.

File: src/tools.cpp [hunk 4]
```diff
@@ -12,9 +12,8 @@
 #include "tools.h"
 
 #include <kodi/General.h>
-#include <p8-platform/util/timeutils.h>
 
-using namespace P8PLATFORM;
+#include <chrono>
```
Comment (AlwinEsch) [inline src/tools.cpp]: As also in header not required here.

File: src/lib/tsreader/platform.h [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+  {
+    s[i] = wc[i];
+    ++i;
+  }
+  if (i < n) s[i] = '\0';
+
+  return (i);
+};
+
+#endif
```
Comment (AlwinEsch) [inline src/lib/tsreader/platform.h:147]: Maybe this not needed, has looked to change without here https://github.com/AlwinEsch/pvr.argustv/commit/cb5b9c56d724de0c252c13e7695eafba96a6613a

Need to check compile and runtime also on Windows and I'm report later.

**Review by AlwinEsch — COMMENTED** [review]
Looks mostly OK, thing only the platform.h not really needed.

Looking currently to become my backend working again.
The connect to there was already OK, only has it not worked with stream (backend self related error) and make it now working again.

File: src/lib/tsreader/platform.h (hunk 5 above)
Comment (phunkyfish) [reply src/lib/tsreader/platform.h:147]: Without would be way better.

**Review by AlwinEsch — APPROVED** [review]
Is good to come in, about the use of inputstream intance within, I must look more detailed how the best to bring (without playback stop :smirk:).

But was good to see that possible :smile:

**Pull request merged**